### PR TITLE
[WIP] Add preliminary support in wholeBodyDynamicsDevice to use BerdySparseMAPSolver

### PR DESCRIPTION
**Pull request by traversaro**
Depends on iDynTree's branch `floatingBerdyPlusPlus` .

### CODE REVIEW

**Comment by traversaro** [comment]
I am afraid that for this we need to add a `devel` branch in codyco-modules .

File: src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 1]
```diff
@@ -269,22 +279,163 @@ bool WholeBodyDynamicsDevice::openEstimator(os::Searchable& config)
...
+
+    iDynTree::FrameIndex imuFrameIndex = deterministicEstimator.model().getFrameIndex(settings.imuFrameName);
+
+
+    iDynTree::AccelerometerSensor linearAccelerometer;
+    linearAccelerometer.setName(settings.imuFrameName);
+    linearAccelerometer.setParentLink(model.getFrameName(model.getFrameLink(imuFrameIndex)));
+    linearAccelerometer.setLinkSensorTransform(model.getFrameTransform(imuFrameIndex));
+    linearAccelerometer.updateIndices(model);
+    ok = ok && sensorsActuallyUsedInEstimation.addSensor(linearAccelerometer);
```
Comment (traversaro) [inline src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: addSensor returns an int.

File: src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 2]
```diff
@@ -269,22 +279,163 @@ bool WholeBodyDynamicsDevice::openEstimator(os::Searchable& config)
...
+
+    iDynTree::ThreeAxisAngularAccelerometerSensor angularAccelerometer;
+    angularAccelerometer.setName(settings.imuFrameName);
+    angularAccelerometer.setParentLink(model.getFrameName(model.getFrameLink(imuFrameIndex)));
+    angularAccelerometer.setLinkSensorTransform(model.getFrameTransform(imuFrameIndex));
+    angularAccelerometer.updateIndices(model);
+    ok = ok && sensorsActuallyUsedInEstimation.addSensor(angularAccelerometer);
```
Comment (traversaro) [inline src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: See https://github.com/robotology/codyco-modules/pull/240/files#r136834708

**Comment by traversaro** [comment]
This PR (adding Berdy-like estimation support to the WBD device that runs on the robot) adds a lot of complexity to wbd, with little added value.
Unless someone is willing to work on this, I think we can decline it and just recover it when somebody will want to work on this. 
cc @DanielePucci

**Comment by DanielePucci** [comment]
@traversaro 
Completely agree. In the long run, probably @prashanthr05 may take upon this, according to his scheduled activities

**Comment by traversaro** [comment]
Ok. Closing the PR, as GitHub remembers, so if we ever need this code, we can just go back to this PR and recover it.

**Pull request closed without merging**